lib/puppet/indirector/rest.rb

```diff
@@ -55,15 +55,17 @@ def self.srv_service
   def self.server
```

Comment:
@joshcooper I looked into setting a `@server` setting here, but it caused a whole lot of spec tests to fail. @jacob brought up the concern that there was no cache invalidation. So that once `@server` is set, it will persist with long running agent daemons, and it'll return an unexpected server because of that. Based on the spec failures, it looks like he has a good point. Did you still want me to investigate making this happen?

Reply:
That's a good point. I think what you have is fine, though during the merge up to later branches it'd be good to perserve the `debug_once` behavior that Charlie added in 179f3e8e5d0cc059ccc4e77a96e1d828b716015d

Reply:
I'll do a manual merge up once this is merged in to make sure we do preserve that